input/blog/2025-07-17-dotnet-cake-cs.md

```diff
@@ -0,0 +1,296 @@
+## Getting Started with Cake.Template
+
+The easiest way to get started will be using the **Cake.Template** package, which provides several templates for different scenarios:
+
+First, install the template package:
+
+```bash
+dotnet new install Cake.Template@5.0.25198.49-beta
+```
+
+### Version Control with global.json
+
+For all scenarios, it's recommended to create a `global.json` file to pin both the .NET SDK and Cake.Sdk versions:
+
+```bash
+dotnet new cakeglobaljson
+```
+
+This creates a `global.json` file that looks like:
+
+```json
+{
+  "sdk": {
+    "version": "10.0.100-preview.6.25358.103"
+  },
+  "msbuild-sdks": {
+    "Cake.Sdk": "5.0.25198.49-beta"
+  }
+}
+```
+
+If you already have a `global.json` file, you can add the `msbuild-sdks` section to control the Cake.Sdk version used.
+
+> **Note**: There's currently a bug in .NET 10, but hopefully by preview 7 you'll also be able to specify the Cake.Sdk version in file-based apps like this:
+> ```csharp
+> #:sdk Cake.Sdk@version
+> ```
+
+### File-based Build Script (Requires .NET 10)
+
+For the new file-based approach that works with `dotnet cake.cs`:
+
+```bash
+dotnet new cakefile --name cake --IncludeExampleProject true
+```
+
+This creates a standalone `cake.cs` file with the `#:sdk Cake.Sdk` directive and example code to clean, build and test a .NET solution. The `--IncludeExampleProject true` flag will create a sample project too (default is false). The resulting file structure will be shown below.
+
+```
+cake.cs
+src
+  │   Example.sln
+  │
+  ├───Example
+  │       Class1.cs
+  │       Example.csproj
+  │
+  └───Example.Tests
+          Example.Tests.csproj
+          UnitTest1.cs
+```
+
+Execute it with:
+```bash
+dotnet cake.cs
+```
+
+### Multi-file File-based Build Script (Requires .NET 10)
+
+For larger projects, you can organize your code into multiple files:
+
+```bash
+dotnet new cakemultifile --name cake
+```
+
+This creates a structure like:
+
+```
+build/
+├── ExceptThisFile.cs
+├── Models.cs
+└── Utilities.cs
+cake.cs
+```
+
+Where `cake.cs` contains similar project as single file but utilizing MSBuild properties `IncludeAdditionalFiles` and `ExcludeAdditionalFiles` to include/exclude additional files example:
+```csharp
+#:sdk Cake.Sdk
+#:property IncludeAdditionalFiles=build/**/*.cs
+#:property ExcludeAdditionalFiles=build/**/Except*.cs
+
+```
+
+### Project-based Build Script
+
+For traditional project-based approach (works with .NET 8, 9, and 10):
+
+```bash
+dotnet new cakeproj --Framework net9.0 --name cake
+```
+
+This creates a `cake` folder with a `cake.csproj` with minimal configuration:
+
+```xml
+<Project Sdk="Cake.Sdk/5.0.25198.49-beta">
+  <PropertyGroup>
+    <TargetFramework Condition="$(TargetFrameworks) == ''">net9.0</TargetFramework>
+    <RunWorkingDirectory>$(MSBuildProjectDirectory)/..</RunWorkingDirectory>
+  </PropertyGroup>
+</Project>
+```
+
+and a `cake.cs` containing minimal example code to clean, build and test and .NET solution.
```

Comment:
Something missing at the beginning of this sentence?

Also `and test and .NET solution.` sounds wrong